src/components/scenes/PasswordLoginScene.tsx

```diff
@@ -182,14 +187,30 @@ export const PasswordLoginScene = (props: Props) => {
       .then(() => {
         logEvent('Pasword_Login')
       })
-      .catch(error => {
+      .catch(async error => {
         if (error != null && error.name === 'OtpError') {
           dispatch({
             type: 'NAVIGATE',
             data: { name: 'otpError', params: { otpAttempt, otpError: error } }
           })
         } else {
           console.log(error)
+
+          const challengeError = asMaybeChallengeError(error)
+          if (challengeError != null) {
+            const result = await Airship.show<'pass' | 'fail' | undefined>(
+              bridge => (
+                <ChallengeModal
+                  bridge={bridge}
+                  challengeError={challengeError}
+                />
+              )
+            )
+            if (result === 'pass') {
+              await handleSubmit()
```

Comment:
You need to pass the `challengeError.challengeId` to `loginWithPassword`. Otherwise we'll just get asked for another CAPTCHA.